### PR TITLE
[Azure Search] Remove InternalsVisibleTo

### PR DESCRIPTION
**Pull request by bitsforthought**
FYI @arv100kri @ishansrivastava90 @updixit @shmed @natinimni @chidozieononiwu @shahabhijeet 

Based on feedback from @weshaggard, I decided to bite the bullet and get rid of `InternalsVisibleTo` in the Azure Search data plane SDK (also, at one point `InternalsVisibleTo` had broken the build due to signing issues, but that has since been resolved).

Here's a summary of the changes:

- Made constructors for facets public (seemed good for customers anyway, since it makes unit testing easier).
- Added test hooks for `SearchContinuationToken` tests. This seemed better than making its members and constructor public, since its whole reason for being is encapsulation.
- Removed the test coverage for `SearchParameters.ToRequest`, `SuggestParameters.ToRequest`, and `AutocompleteParameters.ToRequest` (the latter I also changed from an extension method to an instance method). This test coverage was redundant and unnecessary, given how simple those methods are. It was a better option than making their return types public.
- Added test hooks for creating the SDK's custom `JsonConverters`.
- Factored out `$select=*` and Geo-JSON handling into public methods to make them testable.
- This is incidental to the `InternalsVisibleTo` change, but I renamed the custom `JsonConverter` classes to make them more self-describing.
- Bumped patch version to 9.0.2. I don't think any of these changes are breaking, but if folks spot problems, please let me know. It's probably moot anyway since we have SDK changes pending that will bump our major version number (just not in this PR).

### CODE REVIEW

**Review by natinimni — APPROVED** [review]
LGTM

**Comment by weshaggard** [comment]
@brjohnstmsft perhaps I missed it but I don't see where you removed the InternalsVisibleTo attribute from the project.

**Comment by brjohnstmsft** [comment]
@weshaggard It was in AssemblyInfo.cs, which I deleted.

**Comment by weshaggard** [comment]
>@weshaggard It was in AssemblyInfo.cs, which I deleted.

I missed it. github search isn't the greatest :)

**Review by weshaggard — APPROVED** [review]
Looks reasonable to me.

File: sdk/search/Microsoft.Azure.Search.Common/src/Microsoft.Azure.Search.Common.csproj [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <Description>Common types needed by the Azure Search .NET libraries. This is not the package you are looking for; It is only meant to be used as a dependency.</Description>
     <AssemblyTitle>Microsoft Azure Search Common Library</AssemblyTitle>
     <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
-    <VersionPrefix>9.0.1</VersionPrefix>
+    <VersionPrefix>9.0.2</VersionPrefix>
```
Comment (weshaggard) [inline sdk/search/Microsoft.Azure.Search.Common/src/Microsoft.Azure.Search.Common.csproj]: nit: Might be worth moving the version property into a common props file to remove some redundancy.

Comment (brjohnstmsft) [reply sdk/search/Microsoft.Azure.Search.Common/src/Microsoft.Azure.Search.Common.csproj]: Sure, we'll do that in a subsequent PR

**Pull request merged**